### PR TITLE
board: Adding Adafruit QT PY RP2040 Support

### PR DESCRIPTION
**Pull request by raveious**
Adding a board option for the Adafruit QT PY - RP2040 variant.

Supersedes #63554

### CODE REVIEW

**Comment by raveious** [comment]
@anangl @soburi @yonsch Would you mind reviewing this?

File: boards/arm/adafruit_qt_py_rp2040/seeed_xiao_connector.dtsi [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+ */
+
+/ {
+	xiao_d: connector {
+		compatible = "seeed,xiao-gpio";
+		#gpio-cells = <2>;
+		gpio-map-mask = <0xffffffff 0xffffffc0>;
+		gpio-map-pass-thru = <0 0x3f>;
+		gpio-map
+			= <0 0 &gpio0 29 0>		/* D0 */
```
Comment (yonsch) [inline boards/arm/adafruit_qt_py_rp2040/seeed_xiao_connector.dtsi:14]: Any particular reason this is formatted the way it is? Makes more sense to me to have the commas after the value like it does here:

https://github.com/zephyrproject-rtos/zephyr/blob/8c83185f6e3d5b41ea86aa01bccd82fc07cb1de5/boards/arm/nucleo_f401re/arduino_r3_connector.dtsi#L13-L16

**Review by yonsch — CHANGES_REQUESTED** [review]
I have no familiarity with the boards but it seems OK. I only have one small remark regarding formatting.
Regarding the commits, they will have to be structured differently. On a quick glance I'd say one commit for adding the board and one commit for adding the test overlay. You can have the mutual commit be signed off by both @hlord2000 and yourself.
Two questions, just to make sure:
1. Have you tested this on actual hardware (more specifically, after rebasing it on a fresh main to get the clock control patch)?
2. Does this replace #63554 completely? Can we close it afterwards?

File: boards/arm/adafruit_qt_py_rp2040/seeed_xiao_connector.dtsi (hunk 1 above)
Comment (raveious) [reply boards/arm/adafruit_qt_py_rp2040/seeed_xiao_connector.dtsi:14]: I copied this file from another board that must have had it that way. I was more focused on making sure that the pins were mapped correctly and missed the weird commas. I'll move the commas.

**Comment by raveious** [comment]
> I have no familiarity with the boards but it seems OK. I only have one small remark regarding formatting. Regarding the commits, they will have to be structured differently. On a quick glance I'd say one commit for adding the board and one commit for adding the test overlay. You can have the mutual commit be signed off by both @hlord2000 and yourself. Two questions, just to make sure:
> 
> 1. Have you tested this on actual hardware (more specifically, after rebasing it on a fresh main to get the clock control patch)?
> 2. Does this replace [boards: add support for QT Py RP2040 #63554](https://github.com/zephyrproject-rtos/zephyr/pull/63554) completely? Can we close it afterwards?

I'll restructure all of the commits into 2 commits, the first being the changes of the "boards" directory (signed off by both of us), and the second would be just the change to "tests" directory.

1. Yes, I have tested it on actual hardware. I've tested this with the "hello_world" and "usb_console" samples, and everything works as intended. Is there anything else you would like me to try?
2. Yes, this would replace that PR completely. The other one appeared stale and unmaintained. When making comments on the PR, the author said it was ok to open a new PR with the actions addressed.

**Review by yonsch — APPROVED** [review]
LGTM, thanks!

**Review by soburi — APPROVED** [review]
LGTM

**Pull request merged**